lighthouse-core/audits/seo/hreflang.js

```diff
@@ -82,52 +68,46 @@ class Hreflang extends Audit {
       title: str_(UIStrings.title),
       failureTitle: str_(UIStrings.failureTitle),
       description: str_(UIStrings.description),
-      requiredArtifacts: ['Hreflang', 'URL'],
+      requiredArtifacts: ['LinkElements', 'URL'],
     };
   }
 
   /**
    * @param {LH.Artifacts} artifacts
-   * @param {LH.Audit.Context} context
-   * @return {Promise<LH.Audit.Product>}
+   * @return {LH.Audit.Product}
    */
-  static audit(artifacts, context) {
-    const devtoolsLog = artifacts.devtoolsLogs[Audit.DEFAULT_PASS];
-    const URL = artifacts.URL;
+  static audit(artifacts) {
+    /** @type {Array<{source: string|{type: 'node', snippet: string}}>} */
+    const invalidHreflangs = [];
+
+    for (const link of artifacts.LinkElements) {
+      if (link.rel !== 'alternate') continue;
+      if (!link.hreflang || isValidHreflang(link.hreflang)) continue;
```

Comment:
should ignore if in the body as well?
https://html.spec.whatwg.org/multipage/links.html#body-ok

Reply:
> should ignore if in the body as well?

oh, I guess implicit in the if/else-if below? Mind making it explicit?

Reply:
sure, done